packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx

```diff
@@ -564,15 +565,17 @@ export const DateInstanceValueEditor = observer(
         valueSpecification.genericType.value.rawType.name !==
         PRIMITIVE_TYPE.LATESTDATE
       ) {
-        valueSpecification.values = [
-          format(
-            new Randomizer().getRandomDate(
-              new Date(Date.now()),
-              addDays(Date.now(), 100),
+        runInAction(() => {
```

Comment:
Let's avoid using `runInAction` as this is ad-hoc graph modification, we have a centralized place to do this, why don't we use `instanceValue_changeValues` instead?